wwrmi/api.go

```diff
@@ -824,47 +819,51 @@ func (client *Client) clientLogWorker() {
 	client.locker.Unlock()
 }
 
-
 // TODO cancel http requests
 // see: https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/
 
 // PostLogs sends logs to the API
 func (client *Client) postLogs() (err error) {
 	var req *http.Request
 	var resp *http.Response
-	debugging.DEBUG_OUT("RMI POST %s >>>\n",client.postLogsUrl)
+	debugging.DEBUG_OUT("RMI POST %s >>>\n", client.postLogsUrl)
 	// Client implements io.Reader's Read(), so we do this
 	client.sentBytes = 0
 	req, err = http.NewRequest("POST", client.postLogsUrl, client)
-	//	req.Cancel = c
+	if err != nil {
+		log.MaestroErrorf("Error on POST request: %s\n", err.Error())
+		debugging.DEBUG_OUT("RMI ERROR: %s\n", err.Error())
+		return
+	}
 
-	if err == nil {
-		resp, err = client.client.Do(req)
-		if err != nil {
-			if resp != nil {
-				defer resp.Body.Close()
-			}
-		}
-		debugging.DEBUG_OUT("RMI --> response +%v",resp)
-		if err == nil && resp != nil && resp.StatusCode != 200 {
-			debugging.DEBUG_OUT("RMI bad response - creating error object\n")
-			bodystring, _ := utils.StringifyReaderWithLimit(resp.Body,300)
-			log.MaestroErrorf("RMI: Error on POST request for logs: Response was %d (Body <%s>)",resp.StatusCode,bodystring)
-			err = newClientError(resp)
-		}
-	} else {
-		log.MaestroErrorf("Error on POST request: %s\n",err.Error())		
-		debugging.DEBUG_OUT("RMI ERROR: %s\n",err.Error())
+	resp, err = client.client.Do(req)
+	if resp != nil {
+		defer resp.Body.Close()
+	}
+	if err != nil {
+		log.MaestroErrorf("Error on POST request: %s\n", err.Error())
+		debugging.DEBUG_OUT("RMI ERROR: %s\n", err.Error())
+		return
 	}
-	return
-}
 
+	debugging.DEBUG_OUT("RMI --> response +%v", resp)
+	if resp != nil && resp.StatusCode != 200 {
+		debugging.DEBUG_OUT("RMI bad response - creating error object\n")
+		bodystring, _ := utils.StringifyReaderWithLimit(resp.Body, 300)
+		log.MaestroErrorf("RMI: Error on POST request for logs: Response was %d (Body <%s>)", resp.StatusCode, bodystring)
+		err = newClientError(resp)
+		return
+	}
+	// Read and discard response body
+	_, err = io.Copy(ioutil.Discard, resp.Body)
```

Comment:
Should we handle err != nil here? Same applies to below as well.

Reply:
Doesnt really matter.. We are essentially discarding what we read..

Reply:
What does it mean if that call failed? Should we at least log that?

Reply:
Do you want an print? because error code will be propagated back to the caller anyways with return statement. we use the same return variable `err` to assign the error response

Reply:
Ah ok, I didn't realize that, if it returns the "err" we can leave it as it is.